Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py

```diff
@@ -563,16 +563,18 @@ def register(cls, package, local=False):
         if local:
             if package.name == 'autoinstalled':
                 raise ValueError("local package name 'autoinstalled' is forbidden")
-            libraries = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+            containing_path = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+            libraries = os.path.dirname(containing_path)
             checkout_root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(libraries))))
             for candidate in [
+                containing_path,
                 os.path.join(libraries, package.pypi_name),
                 os.path.join(checkout_root, 'Internal', 'Tools', 'Scripts', 'libraries', package.pypi_name),
             ]:
-                if candidate in sys.path:
-                    return package
                 if not os.path.isdir(os.path.join(candidate, package.name)):
                     continue
+                if candidate in sys.path:
+                    return [package]
```

Comment:
This function will return a package (line 554) or a list of package. Would this make the code that calls it a bit complicated? Or no one cares about the return value?

Reply:
`cls.packages` is a `defaultdict(list)` (line 409), so both calls return a list of `Package`. It is populated a bit below, around line 586.

Reply:
@dewei-zhu It returns a list of packages, as @lauromoura points out, that's always what this function should have done, previous version of the code was buggy in this circumstance.